test/upgradeable/extensions/typedMultiasset.ts

```diff
@@ -0,0 +1,262 @@
+import { ethers, upgrades } from 'hardhat';
+import { expect } from 'chai';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { bn, mintFromMock } from '../../utils';
+import {
+  IERC165,
+  IERC6220,
+  IERC5773,
+  IRMRKExternalEquip,
+  IERC6059,
+  IRMRKTypedMultiAsset,
+  IOtherInterface,
+} from '../../interfaces';
+
+// --------------- FIXTURES -----------------------
+
+async function typeMultiAssetFixture() {
+  const factory = await ethers.getContractFactory('RMRKTypedMultiAssetMockUpgradeable');
+  const typedMultiAsset = await upgrades.deployProxy(factory, ['Chunky', 'CHNK']);
+  await typedMultiAsset.deployed();
+
+  return { typedMultiAsset };
+}
+
+async function nestableTypedMultiAssetFixture() {
+  const factory = await ethers.getContractFactory('RMRKNestableTypedMultiAssetMockUpgradeable');
+  const typedNestableMultiAsset = await upgrades.deployProxy(factory, ['Chunky', 'CHNK']);
+  await typedNestableMultiAsset.deployed();
+
+  return { typedNestableMultiAsset };
+}
+
+async function typedEquippableFixture() {
+  const factory = await ethers.getContractFactory('RMRKTypedEquippableMockUpgradeable');
+  const typedEquippable = await upgrades.deployProxy(factory, ['Chunky', 'CHNK']);
+  await typedEquippable.deployed();
+
+  return { typedEquippable };
+}
+
+async function typedExternalEquippableFixture() {
+  const nestableFactory = await ethers.getContractFactory(
+    'RMRKNestableExternalEquipMockUpgradeable',
+  );
+  const nestable = await upgrades.deployProxy(nestableFactory, ['Chunky', 'CHNK']);
+  await nestable.deployed();
+
+  const equipFactory = await ethers.getContractFactory(
+    'RMRKTypedExternalEquippableMockUpgradeable',
+  );
+  const typedExternalEquippable = await upgrades.deployProxy(equipFactory, [nestable.address]);
+  await typedExternalEquippable.deployed();
+
+  await nestable.setEquippableAddress(typedExternalEquippable.address);
+
+  return { nestable, typedExternalEquippable };
+}
+
+describe('RMRKTypedMultiAssetMockUpgradeable', async function () {
+  let typedMultiAsset: Contract;
+
+  beforeEach(async function () {
+    ({ typedMultiAsset } = await loadFixture(typeMultiAssetFixture));
+
+    this.assets = typedMultiAsset;
+  });
+
+  shouldBehaveLikeTypedMultiAssetInterface();
+  shouldBehaveLikeTypedMultiAsset(mintFromMock);
+
+  describe('RMRKTypedMultiAssetMockUpgradeable get top assets', async function () {
+    let owner: SignerWithAddress;
+    let tokenId: number;
+
+    const resId = bn(1);
+    const resId2 = bn(2);
+    const resId3 = bn(3);
+
+    beforeEach(async function () {
+      ({ typedMultiAsset } = await loadFixture(typeMultiAssetFixture));
+
+      const signers = await ethers.getSigners();
+      owner = signers[0];
+
+      tokenId = await mintFromMock(typedMultiAsset, owner.address);
+    });
+
+    it('can get top asset by priority and type', async function () {
```

Comment:
I wonder why these two are not part of the behavior. Can you check if it is possible?